src/api/spec/models/staging_workflow_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'rails_helper'
+
+RSpec.describe StagingWorkflow, type: :model do
+  let(:project) { create(:project_with_package) }
+  let(:staging_workflow) { create(:staging_workflow_with_staging_projects, project: project) }
+  let(:staging_project) { staging_workflow.staging_projects.first }
+  let(:source_project) { create(:project, name: 'source_project') }
+  let(:target_package) { create(:package, name: 'target_package', project: project) }
+  let(:source_package) { create(:package, name: 'source_package', project: source_project) }
+  let(:bs_request) do
+    create(:bs_request_with_submit_action,
+           target_project: project.name,
+           target_package: target_package.name,
+           source_project: source_project.name,
+           source_package: source_package.name,
+           description: 'BsRequest 1')
+  end
+
+  describe '#unassigned_request' do
+    subject { staging_workflow.unassigned_requests }
+
+    context 'without requests in the main project' do
+      it { expect(subject).to be_empty }
+    end
+
+    context 'with requests but not in staging projects' do
+      before { bs_request }
+
+      it { expect(subject.size).to eq(1) }
+      it { expect(subject.first.description).to eq('BsRequest 1') }
```

Comment:
Why do you test for the description and not for the model?

Reply:
that `subject` contains exactly what you want it to contain: a specific instance of `BsRequest`

Reply:
and only that specific instance. nothing else.

Reply:
Rewriten using the `contain_exactly` matcher.